taxi/forms.py

```diff
@@ -0,0 +1,49 @@
+from django.contrib.auth import get_user_model
+from django.contrib.auth.forms import UserCreationForm
+from django import forms
+from django.core.exceptions import ValidationError
+
+from taxi.models import Driver, Car
+
+
+class DriverLicenseUpdateForm(forms.ModelForm):
+    class Meta:
+        model = Driver
+        fields = ("license_number",)
+
+    def clean_license_number(self):
+        num = self.cleaned_data["license_number"]
+        if (len(num) != 8
+                or not num[:3].isupper()
+                or not num[3:].isdigit()
+                or not num[:3].isalpha()):
+            raise ValidationError(
+                "License Number must consist only of 8 chars,"
+                " first 3 chars must be in uppercase,"
+                " last 3 must be digits")
+        return num
+
+
+class DriverForm(UserCreationForm, DriverLicenseUpdateForm):

```

Comment:
It is weird that the more general `DriverForm` inherits from the more specific `DriverLicenseUpdateForm`. Consider to use `composition` design pattern here. One way to do this is by creating a mixin that contains the `clean_license_number`